src/item_factory.cpp

```diff
@@ -257,12 +257,13 @@ void Item_factory::finalize_item_blacklist()
         for( auto &g : m_template_groups ) {
             g.second->remove_item( e.first );
         }
+
         recipe_dict.delete_if( [&]( recipe &r ) {
-            return r.result == e.first || r.requirements.remove_item( e.first );
+            return r.result == e.first || const_cast<requirement_data &>( r.requirements.obj() ).remove_item( e.first );
         } );
 
         remove_construction_if([&](construction &c) {
-            return c.requirements.remove_item( e.first );
+            return const_cast<requirement_data &>( c.requirements.obj() ).remove_item( e.first );
```

Comment:
The existing implementation of `requirement_data::remove_item` looks subtly broken in that it can only remove a `tool` or `component` but not both